execution_engine/src/core/runtime/mod.rs

```diff
@@ -3345,6 +3346,84 @@ where
         self.gas(cost)?;
         Ok(())
     }
+
+    fn new_dictionary(&mut self, output_size_ptr: u32) -> Result<Result<(), ApiError>, Error> {
+        // Proceed with creating new URefs
+        let new_uref = self.context.new_unit_uref()?;
+
+        // check we can write to the host buffer
+        if let Err(err) = self.check_host_buffer() {
+            return Ok(Err(err));
+        }
+        // create CLValue for return value
+        let new_uref_value = CLValue::from_t(new_uref)?;
+        let value_size = new_uref_value.inner_bytes().len();
+        // write return value to buffer
+        if let Err(err) = self.write_host_buffer(new_uref_value) {
+            return Ok(Err(err));
+        }
+        // Write return value size to output location
+        let output_size_bytes = value_size.to_le_bytes(); // Wasm is little-endian
+        if let Err(error) = self.memory.set(output_size_ptr, &output_size_bytes) {
+            return Err(Error::Interpreter(error.into()));
+        }
+
+        Ok(Ok(()))
+    }
+
+    /// Similar to `read`, this function is for reading from the "local cluster" of global state
+    /// referenced by the uref.
+    fn dictionary_get(
+        &mut self,
+        uref_ptr: u32,
+        uref_size: u32,
+        key_bytes_ptr: u32,
+        key_bytes_size: u32,
+        output_size_ptr: u32,
+    ) -> Result<Result<(), ApiError>, Trap> {
+        if !self.can_write_to_host_buffer() {
+            // Exit early if the host buffer is already occupied
+            return Ok(Err(ApiError::HostBufferFull));
+        }
```

Comment:
why not use the same `if let Err(err) = self.check_host_buffer() {            return Ok(Err(err));        }` pattern as in above method?

Reply:
Addressed